src/ceph-volume/ceph_volume/devices/lvm/batch.py

```diff
@@ -271,39 +319,55 @@ def print_help(self):
             detected_devices=self.get_devices(),
         )
 
-    def report(self):
-        if self.args.format == 'pretty':
-            self.strategy.report_pretty(self.filtered_devices)
-        elif self.args.format == 'json':
-            self.strategy.report_json(self.filtered_devices)
+    def report(self, plan):
+        if self.args.format == 'json':
+            print(json.dumps([osd.report_json() for osd in plan]))
```

Comment:
this is a breaking change (for instance, it makes the ceph_volume module in ceph-ansible failing)

before this patch, `ceph-volume lvm batch --report --format json` used to return something like following :
```python
{'changed': True, 'osds': [{'block.db': {'human_readable_size': '9.00 GB', 'path': 'vg: vg/lv', 'percentage': 50, 'size': 9663676416}, 'data': {'human_readable_size': '9.00 GB', 'path': '/dev/sdb', 'percentage': 100.0, 'size': 9663676416.0}}, {'block.db': {'human_readable_size': '9.00 GB', 'path': 'vg: vg/lv', 'percentage': 50, 'size': 9663676416}, 'data': {'human_readable_size': '9.00 GB', 'path': '/dev/sdz', 'percentage': 100.0, 'size': 9663676416.0}}], 'vg': {'devices': '/dev/nvme0n4, /dev/nvme0n3', 'human_readable_size': '18.00 GB', 'human_readable_sizes': '9.00 GB', 'parts': 2, 'percentages': 50, 'size': 19327352832, 'sizes': 9663676416}, 'vgs': []}
```

now it returns something like this :

```python
[{'data': '/dev/sdb', 'data_size': '10.74 GB', 'block_db': '/dev/nvme0n4', 'block_db_size': '9.00 GB'}, {'data': '/dev/sdz', 'data_size': '10.74 GB', 'block_db': '/dev/nvme0n3', 'block_db_size': '9.00 GB'}]
```

Reply:
https://github.com/ceph/ceph-ansible/pull/5461 fixes this.

Reply:
This will likely cause issues in `DeepSea` and `cephadm` as well.